scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -243,21 +243,35 @@ class Router(formatOps: FormatOps) {
           if (!breakSingleLineAfterClose) Policy.emptyPf
           else decideNewlinesOnlyAfterClose(Split(Newline, 0))(close)
         }
-        def getSingleLineDecision: Policy.Pf =
+        def getClassicSingleLineDecision: Policy.Pf =
           if (newlines > 0) null
           else if (style.activeForEdition_2019_11)
             getSingleLineDecisionFor2019Nov
           else
             getSingleLineDecisionPre2019Nov
 
+        def getSingleLineLambdaDecision: Policy.Pf = {
+          val ok = !style.newlines.alwaysBeforeCurlyBraceLambdaParams &&
+            getSpaceAndNewlineAfterCurlyLambda(newlines)._1
+          if (ok) getSingleLineDecisionFor2019Nov else null
```

Comment:
I am concerned about an increasing number of nulls here. It may be error-prone in the future maintaining. Does `Option` really beats perfomance in this case?

Reply:
probably not. i can try to see if we can change to Option instead, the problem earlier was that Split takes a Policy, so i had to do `policyOpt.orNull` anyway.